org.scala-ide.sdt.core/src/org/scalaide/core/internal/decorators/markoccurrences/ScalaOccurrencesFinder.scala

```diff
@@ -24,28 +22,6 @@ case class Occurrences(name: String, locations: List[IRegion])
 class ScalaOccurrencesFinder(unit: InteractiveCompilationUnit) extends HasLogger {
   import ScalaOccurrencesFinder._
 
-  /* IMPORTANT:
-   * In the current implementation, all access to `indexCache` are confined to the Presentation
-   * Compiler thread (and this is why `indexCache` requires no synchronization policy).
-   */
-  private var indexCache: Option[TimestampedIndex] = None
-
-  private def getCachedIndex(lastModified: Long, currentCompiler: IScalaPresentationCompiler): Option[MarkOccurrencesIndex] = indexCache match {
-    case Some(TimestampedIndex(`lastModified`, WeakReference(index))) if index.global eq currentCompiler => Some(index)
```

Comment:
Well, if we wanted a proper cache, we should at least have used a `SoftReference` here, since `WeakReferences` are completely ignored by the GC and cleared rather quickly. We had cache hit rates well below 80%, at least on my system. What happened for me was that the feature worked exactly when the GC cleared the reference, so that everything was recreated from scratch, but failed otherwise.

As described in the commit message, the test is not enough to ensure consistency... the source is compiled multiple times, into semantically mostly, but not entirely equal trees, resulting in all kinds of headaches. I don't think it's worth the effort to invest any more time in this, unless someone complains, or has a really good idea.